### PR TITLE
imp: add go mod tidy all script

### PR DESCRIPTION
**Pull request by julienrbrt**
## Description

I do not know how split you like your PRs, so added the script I used here (https://github.com/cosmos/ibc-go/pull/4843#discussion_r1348554086) in this PR instead.

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by charleenfei — APPROVED** [review]
thanks so much for splitting this out separately from the autocli pr!!

File: scripts/go-mod-tidy-all.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
```
Comment (DimitrisJim) [inline scripts/go-mod-tidy-all.sh:1]: ah, I see you don't have a makefile rule for this tho yea? Seems like it would be nice to have

Comment (julienrbrt) [reply scripts/go-mod-tidy-all.sh:1]: Sure, any preference qua name?

Comment (DimitrisJim) [reply scripts/go-mod-tidy-all.sh:1]: none! `tidy-all` is what I would go for if pressed.

**Comment by faddat** [comment]
I've been thinking about doing this for some time.  Thank you for this PR!

**Review by chatton — APPROVED** [review]
LGTM thanks

**Pull request merged**